docs/posts/ibis-bench/index.qmd

```diff
@@ -0,0 +1,1313 @@
+    - polars
+---
+
+*The best benchmark is your own workload on your own data*.
+
+## Key considerations
+
+The purpose of this post is to explore some benchmarking data with Ibis. We'll
+compare a few modern query engines, explore the Ibis API as a great choice for
+each of them, and discuss the results.
+
+### The benchmark
+
+:::{.callout-important title="Not an official TPC-H benchmark"}
+This is not an [official TPC-H benchmark](https://www.tpc.org/tpch). We ran a
+derivate of the TPC-H benchmark.
+:::
+
+[The TPC-H benchmark](https://www.tpc.org/tpch) is a benchmark for databases
+and, [increasingly](https://docs.coiled.io/blog/tpch),
+[dataframes](https://pola.rs/posts/benchmarks)!  It consists of 22 queries
+across 8 tables. The queries are designed to test the performance of a database
+(or dataframe!) on a variety of tasks including filtering, aggregation, and
+joins.  SQL queries are defined by the TPC-H benchmark. We run the SQL queries
+and equivalent dataframe code via Ibis and Polars APIs.
+
+The data for the benchmark can be generated at any scale factor, which roughly
+corresponds to the size of the data in memory in gigabytes. For instance, a
+scale factor of 10 would be about 10GB of data in memory.
+
+### The engines, the API, the code
+
+We'll use three modern single-node OLAP engines
+([DuckDB](https://github.com/duckdb/duckdb),
+[DataFusion](https://github.com/apache/datafusion),
+[Polars](https://github.com/pola-rs/polars)) with the Ibis API via two coding
+paradigms (dataframe and SQL). Ibis provides a consistent API across 20+
+backends, including these three. We run [SQL
+code](https://github.com/lostmygithubaccount/ibis-bench/blob/v2.0.0/src/ibis_bench/queries/sql.py)
+through Ibis in addition to [dataframe
+code](https://github.com/lostmygithubaccount/ibis-bench/blob/v2.0.0/src/ibis_bench/queries/ibis.py)
+to get a sense of any overhead in Ibis dataframe code. Note that Ibis dataframe
+code generates SQL for the DuckDB and DataFusion backends and generates
+dataframe code for the Polars backend.
+
+:::{.callout-note title="Honorable mention: chDB" collapse="true"}
+[chDB](https://github.com/chdb-io/chdb) would be another great single-node OLAP
+engine to benchmark. We don't because it's not currently a backend for Ibis,
+though [there has been work done to make it
+one](https://github.com/ibis-project/ibis/pull/8497).
+
+If you're interested in contributing to Ibis, a new backend like chDB could be a
+great project for you!
+:::
+
+9/22 queries for Ibis with the Polars backend fail from [lack of scalar subquery
+support](#failing-polars-queries). Due to this and relatively experimental SQL
+support in Polars, we've opted to run on [the Polars API
+directly](https://github.com/lostmygithubaccount/ibis-bench/blob/v2.0.0/src/ibis_bench/queries/polars.py)
+in this iteration of the benchmark. This is done with the LazyFrames API **and
+no streaming engine** (per the Polars team's recommendation, as the current
```

Comment:
```suggestion
no streaming engine** (per the Polars team's recommendation, as the current
```

Can we link to this recommendation? If so, then I think we can nix the `, as the current streaming engine is being rewritten` bit.